metadata_fetcher/fetchers/flk_fetcher.py

```diff
@@ -0,0 +1,257 @@
+import json
+from .Fetcher import Fetcher
+import requests
+from urllib.parse import urlencode
+import settings
+
+
+class FlkFetcher(Fetcher):
+    BASE_URL: str = "https://api.flickr.com/services/rest/"
+
+    def __init__(self, params: dict[str]):
+        """
+        Parameters:
+            params: dict[str]
+        """
+        super(FlkFetcher, self).__init__(params)
+
+        # Tracks where we're at processing photo requests
+        self.photo_index = 1
+        self.photo_total = 0
+
+        # If `next_url` is a param, we know that this is not the fetch of the
+        # first page, so skip setting those attributes
+        if "next_url" in params:
+            for key in params:
+                setattr(self, key, params[key])
+            return
+
+        self.collection_id = params.get("collection_id")
+        self.harvest_extra_data = params.get("harvest_data").\
+            get("harvest_extra_data")
+        self.per_page = 100
+        self.next_url = self.get_current_url()
+
+    def get_current_url(self) -> str:
+        """
+        Returns the request URL based on the type of collection, photoset or
+        user_photos. Note we increment the page number since Flickr starts
+        with page 1 and internal page iteration starts with page 0.
+
+        Returns: str
+        """
+        if self.is_photoset():
+            return self.get_photoset_request_url(self.harvest_extra_data,
```

Comment:
Can you move the implementation of `self.get_photoset_request_url()` into this function instead of making it a separate function?